index.bikeshed.bs

```diff
@@ -343,7 +357,48 @@ implementation detail. It is not an API that implementors
 provide to web applications. It is used in this document
 only to simplify the algorithm description.
 
-## Response verification algorithms ## {#verification-algorithms}
+## Request verification algorithms ## {#request-verification-algorithms}
+
+### Opting-in  ### {#opt-in-require-sri-for}
+
+Authors may opt a {{Document}} to require SRI metadata be present for 
+some resource types via a <dfn export>require-sri-for</dfn> <a>Content
+Security Policy</a> directive defined by the following ABNF grammar:
+
+<pre dfn-type="grammar" link-type="grammar">
+    directive-name  = "require-sri-for"
+    directive-value = <a grammar>token</a> *( <a>RWS</a> <a>token</a> )
+</pre>
+
+The directive recognizes a number of potential token values:
+
+  * `script` requires SRI for scripts
+  * `style` requires SRI for style sheets
+
+### Parsing `require-sri-for` ### {#parse-require-sri-for}
+
+To parse the |token| list, the user agent MUST use an algorithm equivalent to the following:
+
+1. Let the set of |protected resource types| that require SRI be the empty set.
+
+2. For each token returned by <a>splitting tokens on spaces</a>,
+  if token matches the grammar for <a>require-sri-for</a>,
+  add the token to the set of |protected resource types|. Otherwise, ignore the token.
+
+3. Return the set of |protected resource types|.
+
+### Apply |algorithm| to |request| ### {#apply-algorithm-to-request}
+
+1. Let |protected resource types| be the result of applying [[#parse-require-sri-for]]
+  to the value of the <a>require-sri-for</a> directive.
```

Comment:
I think this needs to be clarified. A few questions come to mind:

1. It's not at all clear what "the `require-sri-for` directive" refers to. I think you probably want to talk about `|request|`'s `client`'s (`responsible document`/`global object`'s) `CSP list`?

2. Once you have a CSP list, it's not clear what you expect the behavior to be in the case of conflicting instructions (e.g. one policy with `require-sri-for script` and another with `require-sri-for style`).

3. Lots of things have a `type` of `script` that you might not intend. For example, do you intend for this to apply to {Web,Shared,Service}Workers as well (e.g. `new Worker` in the document, and `importScript` inside the worker)? If so, SRI probably needs to define that mechanism. Likewise, we treat XSLT as `script-src`: should we treat it as "script" here?